prelude-si/e2e/e2e_test.py

```diff
@@ -0,0 +1,199 @@
+#!/usr/bin/env python3
+"""
+Runs Cypress E2E Suite against a stack
+"""
+import argparse
+import os
+import sys
+from enum import Enum, EnumMeta
+from typing import Any, Dict, List, Union
+import subprocess
+import urllib.request
+import time
+import shutil
+
+# A slightly more Rust-y feeling enum
+# Thanks to: https://stackoverflow.com/a/65225753
+class MetaEnum(EnumMeta):
+
+    def __contains__(self: type[Any], member: object) -> bool:
+        try:
+            self(member)
+        except ValueError:
+            return False
+        return True
+
+
+class BaseEnum(Enum, metaclass=MetaEnum):
+    pass
+
+
+class PlatformArchitecture(BaseEnum):
+    Aarch64 = "aarch64"
+    X86_64 = "x86_64"
+
+
+class PlatformOS(BaseEnum):
+    Darwin = "darwin"
+    Linux = "linux"
```

Comment:
This is a nit, so feel free to ignore. Is there value in detecting this here? Both our README and Nix flake are only compatible with this platform matrix anyway.

Reply:
Fair comment, I was thinking there was a chance it might make it through to this point on a strange variant, or maybe we raw-port the python onto alpine or something as an execution environment.

I.e. by invoke like
```
python3 ./e2e_test.py
```

Reply:
Gotcha. Let's keep it.